frontend/app/login/form.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { signIn } from "next-auth/react";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import * as yup from "yup";
+import { Eye, EyeOff } from "react-feather";
+import Link from "next/link";
+import { Section } from "@/components/section";
+
+interface LoginFormData {
+  email: string;
+  password: string;
+}
+
+const loginUserSchema = yup.object({
+  email: yup.string().trim().required("Email is required").email("Must be a valid email"),
+  password: yup.string().required("Password is required").min(6, "Password must be at least 6 characters long"),
+});
+
+export default function Login() {
+  const [error, setError] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting },
+  } = useForm<LoginFormData>({
+    defaultValues: {
+      email: "",
+      password: "",
+    },
+    resolver: yupResolver(loginUserSchema),
+  });
+
+  const router = useRouter();
+  const onSubmit = async (data: LoginFormData) => {
+    const credentials = {
+      email: data.email,
+      password: data.password,
+    };
+
+    const response = await signIn("login", {
+      ...credentials,
+      redirect: false,
+    });
+    if (response?.error) {
+      setError(response.error);
+    }
+
+    if (response?.ok) {
+      router.push("/");
+    }
+  };
+  const loginHeader = (
+    <div className="form-main-text">
+      <h2>Login</h2>
+    </div>
+  );
+  return (
+    <main className="form-main">
+      <div className="section-wrapper">
+        <Section header={loginHeader}>
+          <>
+            <form
+              className="form"
+              onSubmit={handleSubmit(onSubmit)}
+            >
+              <label htmlFor="email">Email</label>
+              <input
+                id="email"
+                type="email"
```

Comment:
`autocomplete="username"`
https://www.chromium.org/developers/design-documents/form-styles-that-chromium-understands/#email-first-sign-in-flow

Reply:
required